Podverse/PVAuth.swift

```diff
@@ -7,222 +7,104 @@
-////                
-////                if let ownerName = dictResponse["ownerName"] as? String {
-////                    clip.ownerName = ownerName
-////                }
-////                
-////                if let title = dictResponse["title"] as? String {
-////                    clip.title = title
-////                }
-////                
-////                if let startTime = dictResponse["startTime"] as? NSNumber {
-////                    clip.startTime = startTime
-////                }
-////                
-////                if let endTime = dictResponse["endTime"] as? NSNumber {
-////                    clip.endTime = endTime
-////                }
-////                
-////                if let dateCreated = dictResponse["dateCreated"] as? String {
-////                    clip.dateCreated = PVUtility.formatStringToDate(dateCreated)
-////                }
-////                
-////                if let lastUpdated = dictResponse["lastUpdated"] as? String {
-////                    clip.lastUpdated = PVUtility.formatStringToDate(lastUpdated)
-////                }
-////                
-////                if let serverEpisodeId = dictResponse["episodeId"] as? NSNumber {
-////                    clip.serverEpisodeId = serverEpisodeId
-////                }
-////                
-////                CoreDataHelper.saveCoreData(moc, completionBlock: { (saved) in
-////                    dispatch_group_leave(dispatchGroup)
-////                })
-////            }) { (error) -> Void in
-////                print("Not saved to server. Error: ", error?.localizedDescription)
-////                CoreDataHelper.saveCoreData(moc, completionBlock: nil)
-////                }.call()
-//            
-//        }
-//        
-//        if ownedPlaylistsArray.count < 1 && ownedClipsArray.count < 1 {
-//            dispatchGroup.enter()
-//            dispatchGroup.leave()
-//        }
-        
-        dispatchGroup.notify(queue: DispatchQueue.main) {
-            UserDefaults.standard.set(idToken, forKey: "idToken")
-            UserDefaults.standard.set(userId, forKey: "userId")
-//            TODO:
-//            PlaylistManager.shared.getMyPlaylistsFromServer({
-//                PlaylistManager.shared.createDefaultPlaylists()
-//            })
-//            
-            self.delegate?.authFinished()
-            if let cBlock = completionBlock {
-                cBlock()
-            }
+    func syncUserInfoWithServer () {
+        if let idToken = UserDefaults.standard.string(forKey: "idToken"), let userId = UserDefaults.standard.string(forKey: "userId") {
+            Auth0
+                .users(token: idToken)
+                .get(userId, fields: ["nickname"], include: true)
+                .start { result in
+                    switch result {
+                    case .success(let user):
+                        guard let nickname = user["nickname"] as? String else {
+                            return
```

Comment:
Ah good catch. Hmm nickname is provided by default from Auth0, but I think it can be set to an empty string by the user after sign up. I'm thinking we shouldn't be making it a requirement for setting user info at all, since users shouldn't have to provide a nickname to have an account.